tools/mountsnoop.py

```diff
@@ -393,6 +413,8 @@ def main():
     )
     parser.add_argument("--ebpf", action="store_true",
         help=argparse.SUPPRESS)
+    parser.add_argument("-P", "--parent_process", action="store_true",
+        help="also snoop the parent process")
```

Comment:
Why add a flag if the bpf prog is going to unconditionally collect the parent info? Would keep the code simpler to remove the branches for `args.parent_process` too.

I'd say either gate the collection prog side or remove the gate altogether.

Reply:
Thanks.
This is a compatibility issue, which has been discussed with @davemarchevsky and @yonghong-song  before. 
The brief contents are as follows:
Looks like this is the first time field ordering is being changed since mountsnoop's initial commit. Is there any chance someone might be relying on the current order? I'm imagining a situation where someone is parsing the output of mountsnoop and doing pid = fields[1] instead of using first line to determine order.

So we only print the columns when a flag is passed. 
This way default output is not affected.

Nevertheless, if the community strongly recommends deleting the flag, we will follow it.

Reply:
I think the conclusions reached in previous discussion still hold: we should keep output format stable and gate these new columns.

Gating BPF-side collection behind the flag feels like a nice-to-have that doesn't need to be addressed in this PR. I'm in favor of keeping the BPF simple to ease eventual conversion to `libbpf-tools` style and don't think the extraneous 20 bytes of copying will affect perf significantly since this tool is already copying ~400 bytes.